app/domain/authentication/authn_jwt/input_validation/parse_mapped_claims.rb

```diff
@@ -0,0 +1,110 @@
+module Authentication
+  module AuthnJwt
+    module InputValidation
+      # Parse mapped-claims secret value and return a validated mapping hashtable
+      ParseMappedClaims ||= CommandClass.new(
+        dependencies: {
+          validate_claim_name: ValidateClaimName.new(
+            deny_claims_list_value: MANDATORY_CLAIMS_DENY_LIST
+          ),
+          logger: Rails.logger
+        },
+        inputs: %i[mapped_claims]
+      ) do
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ParsingMappedClaims.new(@mapped_claims))
+          validate_mapped_claims_secret_value_exists
+          validate_mapped_claims_value_string
+          validate_mapped_claims_list_values
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ParsedMappedClaims.new(mapping_hash))
+          mapping_hash
+        end
+
+        private
+
+        def validate_mapped_claims_secret_value_exists
+          raise Errors::Authentication::AuthnJwt::MappedClaimsMissingInput if
+            @mapped_claims.blank?
+        end
+
+        def validate_mapped_claims_value_string
+          validate_last_symbol_is_not_list_delimiter
+          validate_array_after_split
+        end
+
+        def validate_last_symbol_is_not_list_delimiter
+          # split ignores empty values at the end of string
+          # ",,ddd,,,,,".split(",") == ["", "", "ddd"]
```

Comment:
Examples should be in test in my opinion and not here

Reply:
I prefer to keep it there. Otherwise nothing justifies the method existence.